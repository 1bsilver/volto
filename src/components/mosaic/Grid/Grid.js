/**
 * Grid component.
 * @module components/Grid
 */

import React, { PropTypes } from 'react';

import { Row } from 'components';

/**
 * Grid component class.
 * @function Grid
 * @param {Object} props Component properties.
 * @param {Object[]} props.rows Rows in the grid.
 * @param {func} props.selectTile Select tile method.
 * @param {func} props.setTileContent Set tile content method.
 * @returns {string} Markup of the row.
 */
const Grid = ({ rows, selectTile, setTileContent }) =>
  <div className="grid">
    {rows.map((row, index) =>
      <Row
        key={index}
        row={index}
        selectTile={selectTile}
        setTileContent={setTileContent}
        tiles={row}
      />
    )}
  </div>;

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
Grid.propTypes = {
  rows: PropTypes.arrayOf(PropTypes.array),
  selectTile: PropTypes.func.isRequired,
  setTileContent: PropTypes.func.isRequired,
};

export default Grid;
