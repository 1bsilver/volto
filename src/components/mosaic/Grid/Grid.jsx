/**
 * Grid component.
 * @module components/Grid
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Grid as UIGrid } from 'semantic-ui-react';

import { Row } from '../../../components';

/**
 * Grid component class.
 * @function Grid
 * @param {Object} props Component properties.
 * @param {Object[]} props.rows Rows in the grid.
 * @param {func} props.selectTile Select tile method.
 * @param {func} props.setHovered Set hovered tile method.
 * @param {func} props.handleDrop Handle drop tile event.
 * @param {func} props.setTileContent Set tile content method.
 * @returns {string} Markup of the row.
 */
const Grid = ({ rows, selectTile, setHovered, handleDrop, setTileContent }) => (
  <UIGrid className="mosaic">
    {rows.map((row, index) => (
      <Row
        row={index}
        selectTile={selectTile}
        setHovered={setHovered}
        handleDrop={handleDrop}
        setTileContent={setTileContent}
        columns={row.columns}
        hovered={row.hovered}
      />
    ))}
  </UIGrid>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
Grid.propTypes = {
  rows: PropTypes.arrayOf(PropTypes.shape({ columns: PropTypes.array }))
    .isRequired,
  selectTile: PropTypes.func.isRequired,
  setHovered: PropTypes.func.isRequired,
  handleDrop: PropTypes.func.isRequired,
  setTileContent: PropTypes.func.isRequired,
};

export default Grid;
