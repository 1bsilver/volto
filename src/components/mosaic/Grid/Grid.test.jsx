import React from 'react';
import renderer from 'react-test-renderer';
import Grid from './Grid';

jest.mock('../Row/Row', () => jest.fn(() => <div />));

test('renders a grid component', () => {
  const component = renderer.create(
    <Grid
      rows={[{ colums: [] }]}
      selectTile={() => {}}
      deleteTile={() => {}}
      setHovered={() => {}}
      handleDrop={() => {}}
      setTileContent={() => {}}
    />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
