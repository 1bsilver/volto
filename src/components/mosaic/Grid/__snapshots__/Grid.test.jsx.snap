// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a grid component 1`] = `
<div
  className="grid"
>
  <div />
  <div />
</div>
`;
